vunit/ui/results.py

```diff
@@ -26,3 +31,20 @@ def merge_coverage(self, file_name, args=None):
         """
 
         self._simulator_if.merge_coverage(file_name=file_name, args=args)
+
+    def get_report(self):
+        """
+        Get a report (dictionary) of tests: status, time and output path
+        """
+        return {
+            test.name: {
+                "status": test._status.name,  # pylint: disable=protected-access
+                "time": test.time,
+                "path": re.sub(
```

Comment:
Thanks for working on this!

In my opinion, path should be the full path to the output directory of individual paths instead of a relative path. That would make working with the results/paths easier, because I think the common case requires the full path anyways.

Reply:
The motivation is to reduce duplication, since all the tests will normally be generated in the same dir. I think that the relevant information in the path is the hash that is appended to the test name, the piece that you cannot know beforehand. With this format, `path name` matches the content of `test_name_to_path_mapping.txt`.

Nonetheless, we can provide a helper function to build the path (`results.get_full_path(val, <location of desired artifact>`). Alternatively, we can provide both the full path and the hash (without the lib name). @kraigher, wdyt? If we provide the *raw* full path, the discussion above about regexp or `os.path` doesn't apply anymore.

Reply:
My thinking is, if we're interested in the output path, then it's because we're trying to do something with the files in there, in which case we want to work with the full path anyways. I can't think of a good usecase where we only want the name of the folder (e.g. `lib.tb_example_many.01.test_fail_eb05d78caf9a3f67d98b38ccc6f078898ace39e9`), but in any case that's easy to get with `os.path.basename`. And that feels cleaner to me than having to build the path with `join(results.output_path, 'test_output', ...)`

Reply:
@kraigher @1138-4EB :
Do you have an opinion about the remaining open topics (full path/relative path, `to_dict` for `TestReport`, ...)? I'd love to get this pull request merged.

Reply:
> I can't think of a good usecase where we only want the name of the folder

When using different docker containers to execute multiple tasks in a pipeline, it is desirable to know the relative paths because absolute location inside the containers/VMs can be very different.

Anyway I reworked the returned object, which is a class now and it provides a helper function:

```py
def post_func(results):
    report = results.get_report()
    print(report.output_path)
    for key, val in report.tests.items():
        print(key, val)
        print(report.get_full_path(val["path"]))
        print(report.get_full_path(val["path"], "output.txt"))

vu.main(post_run=post_func)
```